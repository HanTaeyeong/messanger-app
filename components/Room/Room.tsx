import React from 'react'
import {useRouter} from 'next/router'

function Room() {
  const query = useRouter()?.query
  
  return (
    <div>Room</div>
  )
}

export default Room