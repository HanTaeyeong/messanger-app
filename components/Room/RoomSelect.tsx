import React from 'react'

const mockRooms = [
  { id: '12', name: 'first room' },
  { id: '53', name: 'second room' }
]

function RoomSelect() {

  return (
    <div className='room-list'>
      {mockRooms.map(room =>
        <div className='room-list-item' key={room.id} >{room.name}</div>
      )}
    </div>
  )
}

export default RoomSelect