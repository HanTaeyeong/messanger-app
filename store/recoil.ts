import { atom, selector, useRecoilState } from 'recoil';

export const weekList: String[] = ['Sunday', 'Monday', 'Tuesday', 'Wendsday', 'Thursday', 'Fryday', 'Saturday']

export const testDate = atom<Number>({
    key: 'testDate',
    default: 0,
});

export const testWeekName = selector<String>({
    key: 'testWeekName',
    get: ({ get }) => weekList[(+get(testDate) % 7) | 0] || 'Sunday',
    set: ({ set }) => { }
});
