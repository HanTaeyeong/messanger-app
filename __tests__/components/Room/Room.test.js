import React from 'react'

import Room from '../../../components/Room/Room'
import TestRenderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil'

describe('Login Component test', () => {
    let testComponent;

    beforeEach(() => {
        testComponent = TestRenderer.create(
            <RecoilRoot>
                <Room />
            </RecoilRoot>)
    })

    test('renders well', () => {
        expect(testComponent).toBeTruthy()
        expect(testComponent.toJSON()).toMatchSnapshot();
    })
})
