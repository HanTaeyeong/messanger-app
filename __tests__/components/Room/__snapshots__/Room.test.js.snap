// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component test renders well 1`] = `
<div>
  Room
</div>
`;
